e2e/gui/src/test/java/com/epam/pipeline/autotests/GlobalSearchTest.java

```diff
@@ -116,6 +124,101 @@ public void searchResultCancel() {
     }
 
+    @Test
+    @TestCase(value = {"EPMCMBIBPC-2658"})
+    public void searchForPipeline() {
+        library()
+                .cd(folder)
+                .clickOnDraftVersion(pipeline)
+                .configurationTab()
+                .editConfiguration(defaultProfile, profile ->
+                        profile
+                                .expandTab(EXEC_ENVIRONMENT)
+                                .expandTab(ADVANCED_PANEL)
+                                .clear(NAME).setValue(NAME,configurationName)
+                                .clear(DISK).setValue(DISK,configurationDisk)
+                                .setTypeValue(configurationNodeType)
+                                .sleep(2, SECONDS)
+                                .click(SAVE)
+                                .sleep(2, SECONDS)
+                );
+        home().sleep(2, MINUTES);
+        search()
+                .search(pipeline)
+                .enter()
+                .sleep(2, SECONDS)
+                .ensureAll(GlobalSearchAO.disable, FOLDERS, RUNS, TOOLS, DATA, ISSUES)
+                .ensureAll(enabled, PIPELINES)
+                .ensure(PIPELINES, hasText("3 PIPELINES"))
+                .validateCountSearchResults(3)
+                .hover(SEARCH_RESULT)
+                .openSearchResultItem(pipeline)
+                .ensure(TITLE, hasText(pipeline))
+                .ensure(HIGHLIGHTS, hasText("Found in name"))
+                .ensure(INFO, hasText("draft-"))
+                .parent()
+                .openSearchResultItem("docs/README.md")
+                .ensure(TITLE, hasText("README.md"))
+                .ensure(HIGHLIGHTS, hasText("Found in pipelineName"))
+                .ensure(PREVIEW, hasText("Job definition"))
+                .parent()
+                .openSearchResultItem("config.json")
+                .ensure(TITLE, hasText("config.json"))
+                .ensure(DESCRIPTION, hasText(pipeline))
+                .ensure(HIGHLIGHTS, hasText("Found in pipelineName"))
+                .ensure(PREVIEW, hasText(configurationDisk))
+                .ensure(PREVIEW, hasText(configurationName))
+                .ensure(PREVIEW, hasText(configurationNodeType.substring(0, configurationNodeType.indexOf(" "))))
+                .parent()
+                .moveToSearchResultItem(pipeline, () -> new PipelineLibraryContentAO(pipeline))
+                .assertPipelineName(pipeline);
+        search()
+                .search(pipeline)
+                .enter()
+                .sleep(2, SECONDS)
+                .moveToSearchResultItem("docs/README.md", () -> new DocumentTabAO(pipeline))
+                .shouldContainDocument("README.md");
+        search()
+                .search(pipeline)
+                .enter()
+                .sleep(2, SECONDS)
+                .moveToSearchResultItem("config.json", () -> new PipelineCodeTabAO(pipeline))
+                .ensure(byText("config.json"), visible);
+    }
+
+    @Test
```

Comment:
As I can see from the test case description, "EPMCMBIBPC-2662" depends on "EPMCMBIBPC-2658". Shouldn't we make this test dependent on EPMCMBIBPC-2658?

Reply:
Done